src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -65,6 +65,22 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                     {
                         ConfigureTempApiHashKey(builder, settings.Authentication);
                     }
+
+                    // User-specified configuration file path is considered highest precedence, but does NOT override other configuration sources
+                    string userFilePath = settings.UserProvidedConfigFilePath;
+
+                    if (File.Exists(userFilePath) && Path.GetExtension(userFilePath).ToLower().Equals(".json"))
+                    {
+                        try
+                        {
+                            File.OpenRead(userFilePath).Dispose(); // If this succeeds, we have read-permissions
```

Comment:
This check seems unnecessary. The infrastructure should attempt to read this configuration and fail at some point if we don't have read access.

Reply:
I would not check for permissions here at this time and just call `AddJsonFile`. This is technically an existing problem with our other file-based configuration as well, so I'd rather there be an all-up solution rather than a one-off. Feel free to log an issue for this.